### ISSUE TITLE
itables in marimo notebooks

### ISSUE THREAD

**Issue by liquidcarbon** [mwouts/itables#348]
FYI - does not work directly but can be used in an iframe
https://github.com/marimo-team/marimo/discussions/3192

Cheers!

**Comment by mwouts** [comment]
In the upcoming release v2.4.0 I will make sure that the ITable widget works well in Marimo, and add an entry in the documentation about Marimo.

**Comment by mwouts** [comment]
Hi @liquidcarbon , there is a new release candidate available https://pypi.org/project/itables/2.4.0rc0/.

In Marimo I recommend to use the ITable widget to render tables. Can you give it a try (`pip install itables --pre`) and let me know how it works for you? Thanks!

**Comment by liquidcarbon** [comment]
Looks like it works through anywidget (`from itables.widget import ITable)! I'll need to play a bit more to check the tweaks I normally use (filters on each columns, buttons, layouts).

`itables.show(df)` shows the "loading" logo, with or without `init_notebook_mode`.   This is not expected to work then?

**Comment by mwouts** [comment]
Yes the `show` method will need more work in Marimo. With the new version v2.4.0 my expectation is that the ITable widget becomes as functional as the show method - please keep me posted if that's not the case!

**Comment by liquidcarbon** [comment]
One major difference I just noticed between 2.3.0 and 2.4.0rc1 -
I often include URLs in the dataframe using the function like below

2.3.0 renders them appropriately, but the new version appears to escape tags into `&lt;` - is there a new behavior / new flag that deals with this?

```
def urlify(
    df: pd.DataFrame, column: str, baseurl: str, column_url: str | None = None
) -> pd.DataFrame:
    """Turn a column of a dataframe into a URL."""

    if column_url is None:
        column_url = column
    res = df.copy()
    res[column] = res.apply(
        lambda x: f'<a href="{baseurl}{x[column_url]}">{x[column]}</a>', axis=1
    )
    return res
```

What's a better way to handle it?

**Comment by mwouts** [comment]
Hi @liquidcarbon , yes in version 2.4.0 we're turning to a safer default behavior eg escape HTML by default.

You can still render HTML with this flag : https://mwouts.github.io/itables/options/allow_html.html, and you can also change the default like on the other options.

This way you get an opportunity to confirm that you trust the HTML that will be displayed.
